common/service-support-cdi/src/main/java/io/helidon/common/servicesupport/cdi/CdiExtensionBase.java

```diff
@@ -0,0 +1,484 @@
+     * @return Its class.
+     */
+    public static Class<?> getRealClass(Object object) {
+        Class<?> result = object.getClass();
+        while (result.isSynthetic()) {
+            result = result.getSuperclass();
+        }
+        return result;
+    }
+
+    protected Set<Class<?>> annotatedClasses() {
+        return annotatedClasses;
+    }
+
+    protected Set<Class<?>> annotatedClassesProcessed() {
+        return annotatedClassesProcessed;
+    }
+
+    protected void before(@Observes BeforeBeanDiscovery discovery) {
+        restEndpointInfo = newRestEndpointInfo();
+    }
+
+    /**
+     * Cleans up any data structures created during annotation processing but which are not needed once the CDI container has
+     * started.
+     *
+     * @param adv the {@code AfterDeploymentValidation} event
+     */
+    protected void clearAnnotationInfo(@Observes AfterDeploymentValidation adv) {
+        if (logger.isLoggable(Level.FINE)) {
+            Set<Class<?>> annotatedClassesIgnored = new HashSet<>(annotatedClasses());
+            annotatedClassesIgnored.removeAll(annotatedClassesProcessed());
+            if (!annotatedClassesIgnored.isEmpty()) {
+                logger.log(Level.FINE, () ->
+                        "Classes originally found with selected annotations that were not processed, probably "
+                                + "because they were vetoed:" + annotatedClassesIgnored.toString());
+            }
+        }
+        annotatedClasses.clear();
+        annotatedClassesProcessed.clear();
+    }
+
+    protected R restEndpointInfo() {
+        return restEndpointInfo;
+    }
+
+    /**
+     * Finds an existing or adds a new extension-specific {@code AsyncResponseInfo} for the indicated method.
+     *
+     * @param method the Method for which the AsyncResponseInfo is needed
+     * @return the pre-existing or newly-created instance, if any; null if no existing mapping was found and the factory method
+     * declined to create one
+     */
+    protected A computeIfAbsentAsyncResponseInfo(Method method){
+        Map<Method, A> asyncResponseInfo = restEndpointInfo.asyncResponseInfo();
+        return asyncResponseInfo.computeIfAbsent(method, this::newAsyncResponseInfo);
+    }
+
+    /**
+     * Returns a {@code AsyncResponseInfo} (or subclass) instance describing the async information about the specified method.
+     *
+     * @param method Method to examine for asynchronous behavior
+     * @return AsyncResponseInfo describing the async behavior; null if the method is synchronous
+     */
+    protected abstract A newAsyncResponseInfo(Method method);
+
+    /**
+     * Returns the index in the method's array of parameters, if any, with type {@code AsyncResponse} and annotated with
+     * {@code Suspended}.
+     *
+     * @param m the method to examine
+     * @return the array index of the async parameter, if any; -1 otherwise
+     */
+    protected static int asyncParameterSlot(Method m) {
```

Comment:
Correct, no constructors. Only REST endpoint `Method` instances are ever passed to this method because that's the only place (AFAIK) that `@Suspended` `AsyncResponse` would ever appear.